File: lib/storage.js
const fs = require('fs');
const path = require('path');
const { randomBytes } = require('crypto');

class Storage {
  constructor(baseDir) {
    const base = baseDir || process.env.VERCEL_TEMP || '.data';
    if (!fs.existsSync(base)) fs.mkdirSync(base, { recursive: true });
    this.base = base;
    this.dataFile = path.join(this.base, 'data_lines.txt');
    this.accountsFile = path.join(this.base, 'accounts.json');
    if (!fs.existsSync(this.accountsFile)) fs.writeFileSync(this.accountsFile, JSON.stringify({}), 'utf8');
  }

  appendLines(lines) {
    const stream = fs.createWriteStream(this.dataFile, { flags: 'a', encoding: 'utf8' });
    for (const line of lines) {
      stream.write(String(line).replace(/\r?\n/g, '') + '\n');
    }
    stream.end();
    return lines.length;
  }

  saveAccount(account) {
    const accounts = JSON.parse(fs.readFileSync(this.accountsFile, 'utf8')) || {};
    const key = randomBytes(16).toString('hex');
    accounts[key] = account;
    fs.writeFileSync(this.accountsFile, JSON.stringify(accounts), 'utf8');
    return key;
  }

  getAccount(key) {
    const accounts = JSON.parse(fs.readFileSync(this.accountsFile, 'utf8')) || {};
    return accounts[key] || null;
  }
}

module.exports = Storage;

File: lib/storage.py
import os
import json
import uuid
from threading import Lock
from typing import List, Optional


class Storage:
    """Simple file-backed storage using JSON files in a temporary directory.

    This is designed to be simple and work in serverless environments where
    the filesystem is ephemeral. Data will persist only for the lifetime of
    the instance (cold start). For production, replace with a real DB.
    """

    def __init__(self, base_dir: Optional[str] = None):
        base = base_dir or os.environ.get("VERCEL_TEMP") or ".data"
        os.makedirs(base, exist_ok=True)
        self.base = base
        self.data_file = os.path.join(self.base, "data_lines.txt")
        self.accounts_file = os.path.join(self.base, "accounts.json")
        self.lock = Lock()
        # ensure accounts file exists
        if not os.path.exists(self.accounts_file):
            with open(self.accounts_file, "w", encoding="utf-8") as f:
                json.dump({}, f)

    def append_lines(self, lines: List[str]) -> int:
        with self.lock:
            with open(self.data_file, "a", encoding="utf-8") as f:
                for line in lines:
                    f.write(line.rstrip("\n") + "\n")
            return len(lines)

    def save_account(self, account: str) -> str:
        with self.lock:
            with open(self.accounts_file, "r", encoding="utf-8") as f:
                accounts = json.load(f)
            key = uuid.uuid4().hex
            accounts[key] = account
            with open(self.accounts_file, "w", encoding="utf-8") as f:
                json.dump(accounts, f)
            return key

    def get_account(self, key: str) -> Optional[str]:
        with self.lock:
            with open(self.accounts_file, "r", encoding="utf-8") as f:
                accounts = json.load(f)
            return accounts.get(key)


